### PR TITLE
Bitcoin 0.17.1

### PR DESCRIPTION
**Pull request by luke-jr**
- net-p2p/bitcoin{-qt,d} now ~~default to~~ *can* use the bundled LevelDB, which is modified to use more mmaps for a performance improvement; a system-leveldb USE flag has been added to continue to use the system library instead (*and is enabled by default*).
- dev-libs/leveldb-1.20 appears to be compatible with Bitcoin (I reviewed the relevant diffs), so I am adding it to virtual/bitcoin-leveldb.
- dev-libs/libsecp256k1 has been bumped to 0.1_pre20170928 which corresponds (more or less) to the version bundled with 0.17.1.
- ~ppc64 keywords have been added (this is my development platform).
- Knots has dropped the optional-libevent patch, so libevent is now required.
- We moved debian/rpm packaging stuff to another repo (it tends to get updated post-release), but apparently the .desktop and .protocol files went with it, so I have added these to net-p2p/bitcoin-qt/files for now.

### CODE REVIEW

**Comment by candrews** [comment]
>     * net-p2p/bitcoin{-qt,d} now default to use the bundled LevelDB, which is modified to use more mmaps for a performance improvement; a system-leveldb USE flag has been added to continue to use the system library instead.

I don't like this. Bundled libraries are against Gentoo policy: https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies If there must be such a use flag, please set it to use the system library by default.

File: dev-libs/libsecp256k1/libsecp256k1-0.1_pre20170928.ebuild [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (candrews) [inline dev-libs/libsecp256k1/libsecp256k1-0.1_pre20170928.ebuild]: Please bump to EAPI=7

File: dev-libs/libsecp256k1/libsecp256k1-0.1_pre20170928.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		--with-asm=$asm_opt \
+		--with-bignum=$(usex gmp gmp no) \
+		$(use_enable recovery module-recovery) \
+		--disable-static
+}
+
+src_install() {
+	dodoc README.md
+	emake DESTDIR="${D}" install
+	prune_libtool_files
```
Comment (candrews) [inline dev-libs/libsecp256k1/libsecp256k1-0.1_pre20170928.ebuild]: Banned in EAPI=7, use something like `find "${D}" -name '*.la' -delete || die` instead. See https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html

File: dev-util/bitcoin-tx/bitcoin-tx-0.17.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (candrews) [inline dev-util/bitcoin-tx/bitcoin-tx-0.17.1.ebuild]: Please use EAPI=7

File: dev-util/bitcoin-tx/bitcoin-tx-0.17.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	fi
+
+	eapply_user
+
+	echo '#!/bin/true' >share/genbuild.sh || die
+	mkdir -p src/obj || die
+	echo "#define BUILD_SUFFIX gentoo${PVR#${PV}}" >src/obj/build.h || die
+
+	eautoreconf
+	rm -r src/leveldb src/secp256k1 || die
```
[the remark this answers is not in the document]
Comment (luke-jr) [reply dev-util/bitcoin-tx/bitcoin-tx-0.17.1.ebuild]: IIRC, they need to exist for `eautoreconf` to work, but can be deleted safely afterward.

**Comment by luke-jr** [comment]
> I don't like this. Bundled libraries are against Gentoo policy: https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies If there must be such a use flag, please set it to use the system library by default.

Even though there is likely an annoying performance hit? I hate bundled dependencies too, but I would think the end user experience is more important, no?

**Comment by luke-jr** [comment]
So just to be clear, do you want me to revert it back to system libs only, or simply change the USE default?

**Comment by candrews** [comment]
Just changing the USE flag default would be sufficient :)

**Comment by luke-jr** [comment]
Okay, all the changes requested have been made, with exception to the rm/autoreconf ordering (which I confirmed must be done in that order).

**Comment by candrews** [comment]
merged! Thank you :)

**Pull request closed without merging**